tools/autograd/gen_autograd_functions.py

```diff
@@ -109,7 +109,12 @@
   std::vector<Tensor> grad_result;
   grad_result.reserve(grads.size());
   for (const auto & i : c10::irange(grads.size())) {
-    grad_result.emplace_back(${derivative});
+    const auto& grad = grads[i];
+    if (grad.defined()) {
+      grad_result.emplace_back(${derivative});
+    } else {
+      grad_result.emplace_back(Tensor());
```

Comment:
Is this okay when creating a Tensor whose `.defined()` returns `false`?

Reply:
Yeah its okay to return undefined tensors.